lib/polygeist/Passes/CanonicalizeFor.cpp

```diff
@@ -1521,39 +1521,6 @@ struct MoveWhileDown3 : public OpRewritePattern<WhileOp> {
   }
 };
 
-static bool
-collectEffects(Operation *op,
-               SmallVectorImpl<MemoryEffects::EffectInstance> &effects) {
-  // Collect effect instances the operation. Note that the implementation of
```

Comment:
I removed this function as it seems it was duplicated in Ops.cpp, except the one there has an additional 
```
  // Skip over barriers to avoid infinite recursion (those barriers would ask
  // this barrier again).
  if (isa<BarrierOp>(op))
    return true;
```
at the start.
@wsmoses, according to the comment I felt this should be fine, but if you could take a look.

Reply:
I do think we should check the barrier writing operations here. The reason being that otherwise we may miss a potential write from outside the while loop where this is called from.

Can we may ignoreBarriers an argument?